src/main/java/de/tum/in/www1/artemis/service/NotificationSettingsService.java

```diff
@@ -0,0 +1,105 @@
+package de.tum.in.www1.artemis.service;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+import org.springframework.stereotype.Service;
+
+import de.tum.in.www1.artemis.domain.NotificationOption;
+import de.tum.in.www1.artemis.domain.User;
+import de.tum.in.www1.artemis.domain.enumeration.NotificationType;
+import de.tum.in.www1.artemis.domain.notification.NotificationTitleTypeConstants;
+
+@Service
+public class NotificationSettingsService {
+
+    /**
+     * Finds the deactivated NotificationTypes based on the user's NotificationOptions
+     * @param notificationOptions which should be mapped to their respective NotificationTypes and filtered by activation status
+     * @return a set of NotificationTypes which are deactivated by the current user's notification settings
+     */
+    public Set<NotificationType> findDeactivatedNotificationTypes(Set<NotificationOption> notificationOptions) {
+        Map<NotificationType, Boolean> notificationOptionWitchActivationStatusMap = convertNotificationOptionsToNotificationTypesWithActivationStatus(notificationOptions);
+        Set<NotificationType> deactivatedNotificationTypes = new HashSet<>();
+        notificationOptionWitchActivationStatusMap.forEach((notificationType, isActivated) -> {
+            if (!isActivated) {
+                deactivatedNotificationTypes.add(notificationType);
+            }
+        });
+        return deactivatedNotificationTypes;
+    }
+
+    /**
+     * Converts the provided NotificationType Set to a String Set (representing the titles from NotificationTitleTypeConstants)
+     * @param types Set that should be converted to String
+     * @return the converted String Set
+     */
+    public Set<String> convertNotificationTypesToTitles(Set<NotificationType> types) {
+        return types.stream().map(type -> NotificationTitleTypeConstants.findCorrespondingNotificationTitle(type)).collect(Collectors.toSet());
```

Comment:
```suggestion
        return types.stream().map(NotificationTitleTypeConstants::findCorrespondingNotificationTitle).collect(Collectors.toSet());

```
It seems like this could be replaced with a simpler syntax.

Reply:
I added your change to the code thanks  👍